__tests__/app.test.js

```diff
@@ -47,4 +47,40 @@ describe("/api", () => {
         });
     });
   });
+  describe("/GET api/articles", () => {
+    test(" GET /api/articles/1 returns a status 200 and a single corresponding article", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ body }) => {
+          const article = body.article;
+          expect(article.article_id).toBe(1);
+          expect(article.title).toBe("Living in the shadow of a great man");
+          expect(article.topic).toBe("mitch");
+          expect(article.author).toBe("butter_bridge");
+          expect(article.body).toBe("I find this existence challenging");
+          expect(article.created_at).toBe("2020-07-09T20:11:00.000Z");
+          expect(article.votes).toBe(100);
+          expect(article.article_img_url).toBe(
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+          );
+        });
+    });
+    test("GET /api/articles/banana returns a 404 and 'not found' as parameter is invalid", () => {
+      return request(app)
+        .get("/api/articles/hello")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.message).toBe("not found");
+        });
+    });
+    test("GET /api/articles/100 returns a 200 and empty object as parameter is valid but doesn't exist in database", () => {
+      return request(app)
+        .get("/api/articles/100")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body).toEqual({});
+        });
```

Comment:
Nice thinking on including some error handling here, however lets rethink the situations in the last two tests.
If I ask for article 100, and it doesn't exist, this should be a 'not found' error and relevant code.
If I ask for article 'banana', this isn't a valid request and should be communicated as such. Have a think about which two codes we'd need in these situations and allow your error handlers to pick these up 👍

Reply:
Hi Rich! Thank you for the feedback. Normally I would have allowed PSQL to pick up the error and logged 100 as a 404, but after the lecture today with Liam I was left a bit confused, as he said that technically 100 is correct, but its just not in the database yet but could be at some point, so returning an empty object would be the right thing to do? On all previous sprints I've used psql so happy to change back to this!

Reply:
PSQL won't pick up 100, but it should pick up 'banana'. For 100 you will simply get back an empty array, so if we get an empty array back we know that means the article hasn't been found and can handle it accordingly :)